### ISSUE TITLE
there is some error on the result of patchtst

### ISSUE THREAD

**Issue by analytistic** [tsinghua-fib-lab/UoMo#2]
我们在自己的框架上做了nanjing short 48-16的实验，数据预处理代码如下，使用了minmax 归一化，并放缩到【-1 ，1】，请问下面的数据处理代码和这个仓库的有区别吗，我们patchtst 训练10个epoch 在test数据集上复现的结果是
'mae': '0.01309', 'mse': '0.0006252', 'rmse': '0.025'

       data = self._load_data()

        data_dict = {
            "train": ('X_train','train'),
            "eval": ('X_val','val'),
            "test": ('X_test','test'),
        }

        train_timeseries = np.array(data[data_dict["train"][0]][0])
        clip_datas_train = np.percentile(train_timeseries, 99.99)
        train_timeseries = np.clip(train_timeseries, 0, clip_datas_train) / clip_datas_train

        if processor is not None:
            if inspect.isclass(processor):
                self.processor = processor(
                    scale=self.scale,
                    scaler_type=self.scaler_type,
                )
                self.processor.fit(train_timeseries.reshape(-1, 1))
            else:
                self.processor = processor
        else:
            self.processor = None

        
        self.timeseries = np.array(data[data_dict[self.mode][0]][0])
        self.timeseries = np.clip(self.timeseries, 0, clip_datas_train) / clip_datas_train
        self.timeseries = self.timeseries.reshape(self.timeseries.shape[0], self.timeseries.shape[1],     self.timeseries.shape[2]*self.timeseries.shape[3])
        self.timestamp = np.array(data['timestamps'][data_dict[self.mode][1]])

        
 
    def __len__(self):
        return len(self.timeseries)

    def __getitem__(self, index):
        inputs = self.timeseries[index][:self.seq_len]
        labels = self.timeseries[index][self.seq_len:self.seq_len+self.pred_len]
        if self.processor is not None:
            data = self.processor(
                timeseries=inputs.reshape(-1, 1),
                labels=labels,
                scale=self.scale,
                return_tensors="pt"
            )
            return dict(
                timeseries=data['timeseries'].reshape(inputs.shape),
                labels=data["labels"].reshape(labels.shape),
            )
        else:
            return dict(
                timeseries=torch.tensor(inputs, dtype=torch.float),
                labels=torch.tensor(labels, dtype=torch.float),
            )
